File: Search Insert Position - Leetcode 35/Search Insert Position - Leetcode 35.ts
// Solution 1: T - O(log n), S - O(1)
const searchInsert = (nums: number[], target: number) => {
  const n = nums.length;
  let left = 0,
    right = n - 1;

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);
    if (nums[mid] === target) {
      return mid;
    } else if (nums[mid] > target) {
      right = mid - 1;
    } else {
      left = mid + 1;
    }
  }

  return left;
};

File: Search Insert Position - Leetcode 35/Search Insert Position - Leetcode 35.java
class Solution {
    public int searchInsert(int[] nums, int target){
        int n = nums.length();
        int left = 0, right = n - 1;

        while (left <= right)
        {
            int mid = Math.floor((left + right)/2);
            if(nums[mid] === target)
            {
                return mid;
            } else if (nums[mid] < target)
            {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }

        return left;
    }
}